samples/src/main/java/com/example/retrofit/MultipleDeserializedTypes.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright (C) 2017 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.retrofit;
+
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.concurrent.Executor;
+import okhttp3.Request;
+import okhttp3.ResponseBody;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import retrofit2.CallAdapter;
+import retrofit2.Converter;
+import retrofit2.Retrofit;
+import retrofit2.converter.moshi.MoshiConverterFactory;
+import retrofit2.http.GET;
+
+/**
+ * An example {@link CallAdapter} for returning different deserialized values for 202 responses than
+ * other successful responses.
+ */
+final class MultipleDeserializedTypes {
+  public interface Call<R, T> extends Cloneable {
+    final class Response<R, T> {
+      /**
+       * Null iff the response is 202.
+       */
+      public final retrofit2.Response<R> response;
+
+      /**
+       * Not null iff the response is 202.
+       */
+      public final retrofit2.Response<T> accepted;
```

Comment:
gross to reuse this type for both responses. unsure if it's worth having a marker interface or something else.